content/intermediate/241_pixie/using_pixie/mysql_data.md

```diff
@@ -0,0 +1,37 @@
+---
+title: "Inspect the MySQL Request"
+date: 2021-5-01T09:00:00-00:00
+weight: 24
+draft: false
+---
+
+From the demo app’s YAML file, we know that the catalog service talks to a MySQL database. Let’s inspect the `catalog` service's mysql requests to see if we can get more information about the type of database connection error.
+
+Select the `px/mysql_data` script from the script drop-down menu. This script shows all of the mysql requests Pixie has traced in the cluster. Let’s filter these requests.
+
+Modify the script’s `start_time` to `-30m` or any window that will include when you [triggered the bug](/intermediate/241_pixie/prereqs/#trigger-the-microservices-application-bug) in the Sock Shop app.
+
+Open the script editor using `ctrl+e` (Windows, Linux) or `cmd+e` (Mac).
+
+On line 34, add the following line to filter the mysql requests to just those with errors:
+
+```bash
+# Filter requests to only include those with an error code.
+df = df[df.resp_status == 3]
```

Comment:
I don't have any 3 responses, and I have triggered the multiple search symptom.

Reply:
@jlbutler, if you walk away from your computer for a bit, then head back. the "30 min" window will miss the older errors.

maybe it pays to add a blurb text after the screenshot, like we did in "observe service errors", stating:


> If you don’t see a SQL error, increase the time window or manually trigger the bug again in the Sock Shop web application. Make sure to re-run the script afterwards.

Reply:
@jlbutler - did changing the `start_time` window show the request with error?

Reply:
Unfortunately no, the first thing I did was retrigger, and I also widened the scope on time. I didn't see any responses == 3. I can reproduce the issue with multiple filters returning no results, but not seeing those responses.

Reply:
I tried to reproduce this this morning and I do indeed have a failure and could finish. So, I don't think there's anything wrong here (which we already knew).

I also realized that what I saw last night was an error, rather than just an empty response. I unfortunately didn't screen shot it. I will try and reproduce... my guess is that I had an errant character in the script edit window and this is all a wild goose chase.

We can close this comment out, thanks!